core-bundle/src/Image/PictureFactory.php

```diff
@@ -182,23 +182,25 @@ private function createConfig($size): array
                     $formatsString = implode(';', StringUtil::deserialize($imageSizes->formats, true));
                     $formats = [];
 
-                    foreach (explode(';', $formatsString) as $format) {
-                        [$source, $targets] = explode(':', $format, 2);
-                        $targets = explode(',', $targets);
-
-                        if (!isset($formats[$source])) {
-                            $formats[$source] = $targets;
-                            continue;
-                        }
+                    if ('' !== $formatsString) {
+                        foreach (explode(';', $formatsString) as $format) {
+                            [$source, $targets] = explode(':', $format, 2);
```

Comment:
How about adding the check here? Trailing `;` would then not fail either.

```suggestion
                            if ($format === '') {
                                continue;
                            }
                            [$source, $targets] = explode(':', $format, 2);
```

Reply:
That would never make a difference, only create a useless loop. I don't see how there could be a trailing `;`?

Reply:
```php
foreach (array_filter(explode(';', $formatsString)) as $format {
    // …
}
```
? :grin: